graviti/schema/pyarrow.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 Graviti. Licensed under MIT License.
+#
+"""The PyArrow extension types to represent PortexType."""
+
+import json
+from typing import TYPE_CHECKING, Any, Dict, Type, TypeVar
+
+import pyarrow as pa
+
+if TYPE_CHECKING:
+    # https://pylint.pycqa.org/en/latest/technical_reference/c_extensions.html
+    from pyarrow.lib import DataType
+
+
+class _ExtensionBase(  # pylint: disable=too-few-public-methods
+    pa.ExtensionType  # type: ignore[misc]
+):
+    """The base class for the PyArrow representation of PortexType types.
+
+    Arguments:
+        portex_type_name: The PortexType name.
+        storage_type: The corresponding PyArrow type.
+        nullable: Whether the PortexType is nullable.
+        kwargs: Other keyword arguments to init the PortexType.
+
+    """
+
+    _T = TypeVar("_T", bound="_ExtensionBase")
+
+    _PYARROW_PACKAGE_NAME: str
```

Comment:
we can use `typing.ClassVar` to mark the class variable

Reply:
Why PYARROW package? Should it be PORTEX package?

Reply:
Because it's the extension name in pyarrow?